.azure-pipelines/azure-pipelines-base.yml

```diff
@@ -69,13 +69,5 @@ jobs:
     displayName: 'Publish Test Results'
 
   - script: |
-      for i in {1..5}
-      do
-          python -m codecov --token $(CODECOV_TOKEN) --name "$(NAME)" --required -v
-          if [ "$?" -eq 0 ]
-          then
-              break
-          fi
-          sleep 90
-      done
+      bash <(curl -s https://codecov.io/bash) -n "$(NAME)" -v -C "$(SYSTEM_PULLREQUEST_SOURCECOMMITID)"
```

Comment:
Just curious. If their API is failing or getting rate limited how will using bash help? Or am I mistaking original issue?

Reply:
Codecov roadmap is to make bash uploader the preferred (and if I understood correctly the only) option. As of now that both python and bash uploaders work, using the bash one has two main advantages: it automatically tries to reupload when there are connection errors and it does not require token for public azure pipelies projects.

Reply:
We would not need the manual reupload loop anymore nor we'd need to use the token (and PRs from forks would be able to upload coverage info). I am currently waiting for a bug in the tokenless upload to be solved (this is why there are 2 equivalent PRs, one from a branch and this one from my fork).